src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -462,6 +477,47 @@ public void Dispose()
             _externalCts?.Dispose();
         }
 
+        /// <summary>
+        /// NuGet LogLevel -> MSBuild verbosity
+        /// </summary>
+        private static MSBuildVerbosityLevel GetMSBuildLevel(LogLevel level)
+        {
+            switch (level)
+            {
+                case LogLevel.Error:
+                    return MSBuildVerbosityLevel.Quiet;
+                case LogLevel.Warning:
+                    return MSBuildVerbosityLevel.Minimal;
+                case LogLevel.Minimal:
+                case LogLevel.Information:
+                    return MSBuildVerbosityLevel.Normal;
+                case LogLevel.Verbose:
+                    return MSBuildVerbosityLevel.Detailed;
+                default:
+                    return MSBuildVerbosityLevel.Diagnostic;
+            }
+        }
+
+        /// <summary>
+        /// MSBuild verbosity -> NuGet LogLevel
+        /// </summary>
+        private static LogLevel GetLogLevel(MSBuildVerbosityLevel level)
+        {
+            switch (level)
+            {
+                case MSBuildVerbosityLevel.Quiet:
```

Comment:
This seems wrong because no warnings will be displayed. Same reasoning as above.